### ISSUE TITLE
OpenSearch: customEndpoint's CNAME should point to Dual-Stack endpoint when OpenSearch is using Dualstack

### ISSUE THREAD

**Issue by schabe77** [aws/aws-cdk#30480]
### Describe the feature

When an OpenSearch domain is created with a customEndpoint, a CNAME is created pointing to the 'Domain endpoint (IPv4)' (https://[DOMAIN_ID].[REGION].es.amazonaws.com). If the OpenSearchDomain supports DualStack, then the customEndpoint should point to 'Domain endpoint v2 (dual stack)' (https://[DOMAIN_ID].aos.[REGION].on.aws).

### Use Case

We are trying to get rid of IPv4 addresses. During the transition, it would be great to have everything running in dual-stack mode, and later in pure IPv6. Unfortunately, OpenSearch's custom endpoint doesn't currently support dual stack.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.144.0

### Environment details (OS name and version, etc.)

macOS 14.5 23F79 / openjdk version "21.0.3" 2024-04-16

**Comment by pahud** [comment]
Makes sense to me. Can you share your CDK code snippet on how you create `OpenSearchDomain` that supports DualStack?

**Comment by schabe77** [comment]
here a code snippet only containing the customEndpoint and the DualStack IP address type:
```
software.amazon.awscdk.services.opensearchservice.Domain.Builder.create(this, "OpensearchDomain")
        .customEndpoint(CustomEndpointOptions.builder()
                .domainName(properties.getCustomDomain() + "." + CONFIGURATION.getHostedZoneAttributes().getZoneName())
                .hostedZone(properties.getZone())
                .certificate(properties.getCertificate())
                .build())
        .ipAddressType(IpAddressType.DUAL_STACK)
        ...
        .build();
```
